packages/components/src/custom-select-control/index.tsx

```diff
@@ -35,6 +42,15 @@ function applyOptionDeprecations( {
 	};
 }
 
+function getDescribedBy( currentValue: string, describedBy?: string ) {
+	if ( describedBy ) {
+		return describedBy;
+	}
+
+	// translators: %s: The selected option.
+	return sprintf( __( 'Currently selected: %s' ), currentValue );
+}
```

Comment:
This function copies the original logic, which looked like this:

https://github.com/WordPress/gutenberg/blob/9bb75b35ec0daa46e8cd60c46619f522c7edd453/packages/components/src/custom-select-control/index.js#L130-L141

I removed the case for no selection, because that won't happen in this Ariakit-based version.